### PR TITLE
Fix Ignitor bug

### PR DESCRIPTION
**Pull request by pranavkm**
The implementation of the Ignitor has a few differences compared to BrowserRenderer
which results in exceptions. This change addresses these issue. In addition, this
adds a minimal sample that makes it easier to test Ignitor without having to
package it.

### CODE REVIEW

File: src/Components/Ignitor/src/ElementHive.cs [hunk 1]
```diff
@@ -259,7 +259,7 @@ private int InsertFrame(RenderBatch batch, ContainerNode parent, int childIndex,
 
                 case RenderTreeFrameType.Region:
                     {
-                        return InsertFrameRange(batch, parent, childIndex, frames, frameIndex + 1, frameIndex + CountDescendantFrames(frame));
+                        return InsertFrameRange(batch, parent, childIndex, frames, frameIndex + 1, frameIndex + frame.RegionSubtreeLength);
```
Comment (pranavkm) [inline src/Components/Ignitor/src/ElementHive.cs:262]: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Web.JS/src/Rendering/BrowserRenderer.ts#L212

File: src/Components/Ignitor/src/ElementHive.cs [hunk 2]
```diff
@@ -334,12 +335,21 @@ private void InsertElement(RenderBatch batch, ContainerNode parent, int childInd
                 }
                 else
                 {
+                    parent.InsertLogicalChild(newElement, childIndex);
```
Comment (pranavkm) [inline src/Components/Ignitor/src/ElementHive.cs:338]: Per https://github.com/dotnet/aspnetcore/commit/4b943417a7d3e266d3ac998b3ec7e8ffff429910#diff-5341853367d8cb5a9b51579d92422ca1

File: src/Components/Ignitor/src/BlazorClient.cs [hunk 3]
```diff
@@ -35,8 +36,10 @@ public BlazorClient()
             });
         }
 
-        public TimeSpan? DefaultConnectionTimeout { get; set; } = TimeSpan.FromSeconds(20);
-        public TimeSpan? DefaultOperationTimeout { get; set; } = TimeSpan.FromMilliseconds(500);
+        public TimeSpan? DefaultConnectionTimeout { get; set; } = Debugger.IsAttached ?
```
Comment (captainsafia) [inline src/Components/Ignitor/src/BlazorClient.cs:39]: Why is the timeout different between debugger attached and not?

Unrelated: why is this nullable?

**Review by captainsafia — APPROVED** [review]
LGTM -- posted a small question inline.

**Comment by mkArtakMSFT** [comment]
@pranavkm does this have a related Issue? If not, please assign a milestone at least (Issue is preferrable though).

File: src/Components/Ignitor/src/BlazorClient.cs (hunk 3 above)
Comment (pranavkm) [reply src/Components/Ignitor/src/BlazorClient.cs:39]: It was annoying to have these timeouts kick in while I was debugging the app. We do this sort of thing in other places where timeouts are shut off if you're debugging.

Comment (pranavkm) [reply src/Components/Ignitor/src/BlazorClient.cs:39]: I didn't super dig in to why it was null. But does seem like something that could be changed.

**Comment by pranavkm** [comment]
@mkArtakMSFT no particular issue, just something I ran in to when investigating a scenario for customers that use Ignitor for their testing.

**Pull request merged**